src/cosalib/aliyun.py

```diff
@@ -9,3 +16,100 @@ def remove_aliyun_image(aliyun_id, region):
                      '--force'])
     except SystemExit:
         raise Exception("Failed to remove image")
+
+
+@retry(stop=stop_after_attempt(3))
+def aliyun_run_ore_replicate(build, args):
+    build.refresh_meta()
+    aliyun_img_data = build.meta.get('aliyun', [])
+    if len(aliyun_img_data) < 1:
+        raise SystemExit(("buildmeta doesn't contain source images. "
+                         "Run buildextend-aliyun first"))
+
+    if not args.regions:
+        args.regions = subprocess.check_output([
```

Comment:
nit: If no regions means all regions it may be a good idea to log it to the user.

Reply:
:+1: fixing in the next push.